corehq/apps/data_dictionary/views.py

```diff
@@ -90,6 +92,10 @@ def data_dictionary_json(request, domain, case_type_name=None):
                     ),
                     'name': prop.name,
                     'deprecated': prop.deprecated,
+                    'is_geo_case_property': (
+                        data_validation_enabled
```

Comment:
why do we need to check for data_validation_enabled here?
Infact do we need to check for the geospatial flag enabled here?

Reply:
Since the Data Dictionary "Data Type" column requires the `CASE_IMPORT_DATA_DICTIONARY_VALIDATION` feature flag to be enabled, we simply default this property to `False` if it is not enabled. I agree we should probably also check that the `GEOSPATIAL` flag is enabled as well. I'll make a code change to check for both flags.

Reply:
[3cf8319](https://github.com/dimagi/commcare-hq/pull/33426/commits/3cf8319c98cc7355c99a6fae1834ba5dc3e49ade)

Reply:
So what happens if a domain has geospatial feature flag enabled but not `CASE_IMPORT_DATA_DICTIONARY_VALIDATION`?
Should not the validation we want to add be irrespective of this feature flag?

Reply:
This situation could be possible if the domain sets the data types of their case properties with the flag enabled, and then turn the flag off afterwards. I don't know how likely this would be however, so I think in staying cautious we can rather show the modal irrespective of whether the domain can edit the "Data Type" column. This would mean only checking if the `GEOSPATIAL` flag is enabled.

Reply:
help me understand,
if the other feature flag isn't enabled, the column to change the type won't be present on the page? So, one can't really change the data type.
Which is okay since we are not recommending them to update the type but want to confirm what the user would see.

Is the property being used for geospatial setting, expected to be of a specific type?

Reply:
That is correct. When the `CASE_IMPORT_DATA_DICTIONARY_VALIDATION` flag is disabled then the DD will look as follows:
![Screenshot from 2023-09-06 12-42-37](https://github.com/dimagi/commcare-hq/assets/122617251/cfc2afe7-e6f6-4549-9615-f7cfc4850ac0)

The user won't be able to view/edit the data type of a case property with the validation flag disabled. We only show a dropdown of case properties that have a GPS data type in the geospatial settings, so I don't think it would be very likely for users to be in a situation where they are deprecating a case property that is being used by the geospatial settings if they are unable to set a case property's data type to GPS.

Given the above, I'm happy to remove checking for the `CASE_IMPORT_DATA_DICTIONARY_VALIDATION` flag and then users will still get the warning modal even if they can't update/change the data type. This might be possible if the user has previously set a case property's data type, but has since had the validation flag disabled.

Reply:
> Given the above, I'm happy to remove checking for the CASE_IMPORT_DATA_DICTIONARY_VALIDATION flag and then users will still get the warning modal even if they can't update/change the data type. This might be possible if the user has previously set a case property's data type, but has since had the validation flag disabled.

That sounds good to me @zandre-eng , thank you!